### PR TITLE
Fix: Remove overlapping intervals in old snapshots superseded by a new unpaused snapshot with the effective date set

### PR DESCRIPTION
**Pull request by izeigerman**

### CODE REVIEW

File: sqlmesh/core/snapshot/definition.py [hunk 1]
```diff
@@ -708,6 +708,14 @@ def is_materialized(self) -> bool:
     def is_paused(self) -> bool:
         return self.unpaused_ts is None
 
+    @property
+    def normalized_effective_from_ts(self) -> t.Optional[int]:
+        return (
+            to_timestamp(self.model.cron_floor(self.effective_from))
+            if self.effective_from
+            else None
```
Comment (tobymao) [inline sqlmesh/core/snapshot/definition.py:716]: just return 0 here?

Comment (tobymao) [reply sqlmesh/core/snapshot/definition.py:716]: 0 is falsey

Comment (izeigerman) [reply sqlmesh/core/snapshot/definition.py:716]: I know, right. But I felt like 0 is a bad default for a timestamp value

**Review by tobymao — APPROVED** [review]

File: sqlmesh/core/snapshot/definition.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/snapshot/definition.py:716]: who cares, it's millis

Comment (izeigerman) [reply sqlmesh/core/snapshot/definition.py:716]: eg. why not -1 then?

Comment (tobymao) [reply sqlmesh/core/snapshot/definition.py:716]: -1 is not falsey

Comment (izeigerman) [reply sqlmesh/core/snapshot/definition.py:716]: I know, I just meant that it's a perfectly reasonable candidate for a default value. I'd rather be as specific as possible in public interface.

**Pull request merged**